### PR TITLE
Footer

### PR DESCRIPTION
**Pull request by SumairaRasheed**
Created Footer Section
Changes in Header Section

### CODE REVIEW

**Comment by haris-01** [comment]
We need to make sure that the PR should only contain changes about which the branch is about. in this case, changes should be only related to the Footer but it also contains changes in the header as wel.

File: components/Footer/NavigationCard.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import Link from "next/link";
+import { NavLink } from "@/src/types/footerTypes";
```
Comment (haris-01) [inline components/Footer/NavigationCard.tsx:3]: import type please

File: components/Footer/TechnologiesCard.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { LinkItem } from "@/src/types/footerTypes";
```
Comment (haris-01) [inline components/Footer/TechnologiesCard.tsx:2]: import type please

File: components/Footer/TechnologiesCard.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    <div className="bg-stone-200 p-4 md:p-6 space-y-4">
+      <div className="text-blue-900 font-bold text-lg md:text-lg">
+        {linkHeading.heading}
+      </div>
+      <div className="text-black">
+        <div className="flex flex-col md:flex-row md:space-x-4 flex-wrap">
+          <ul className="space-y-3 lg:border-l border-dashed border-gray-500 pl-4 lg:text-xs">
+            {link.map((tech) => {
+              const href =
+                tech.page.slug === "home" ? "/" : `/${tech.page.slug}`;
```
Comment (haris-01) [inline components/Footer/TechnologiesCard.tsx:24]: i've seen this on mutiple locations, can we create a funtion for this?

File: components/Footer/TechnologiesCard.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  link,
+}) => {
+  return (
+    <div className="bg-stone-200 p-4 md:p-6 space-y-4">
+      <div className="text-blue-900 font-bold text-lg md:text-lg">
+        {linkHeading.heading}
+      </div>
+      <div className="text-black">
+        <div className="flex flex-col md:flex-row md:space-x-4 flex-wrap">
+          <ul className="space-y-3 lg:border-l border-dashed border-gray-500 pl-4 lg:text-xs">
```
Comment (haris-01) [inline components/Footer/TechnologiesCard.tsx:21]: we should create. reusable component out of this ui component with classes already assigned. this with reduce complexity, make it easier to use and better readable.

File: components/Footer/TechnologiesCard.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            {link.map((tech) => {
+              const href =
+                tech.page.slug === "home" ? "/" : `/${tech.page.slug}`;
+              return (
+                <li key={tech.id}>
+                  <Link href={href} passHref>
+                    <span className="hover:text-blue-500 hover:underline">
```
Comment (haris-01) [inline components/Footer/TechnologiesCard.tsx:28]: we should create a resuable component for links as well.

File: components/Footer/index.tsx [hunk 6]
```diff
@@ -1,5 +1,76 @@
...
+import footer from "@/src/globalData/footer.json";
+import Image from "next/image";
+import { FaLinkedin } from "react-icons/fa";
+import { BiLogoInstagramAlt } from "react-icons/bi";
+import { IoLogoFacebook } from "react-icons/io5";
+import CompanyCard from "./CompanyCard";
+import NavigationCard from "./NavigationCard";
+import TechnologiesCard from "./TechnologiesCard";
+import ContactInfoCard from "./ContactInfoCard";
+import { FooterData } from "@/src/types/footerTypes";
```
Comment (haris-01) [inline components/Footer/index.tsx:11]: import type please

File: components/Footer/index.tsx [hunk 7]
```diff
@@ -1,5 +1,76 @@
...
 
-export default function Footer() {
-  return <div>Footer</div>;
-}
+const Footer: React.FC = () => {
+  const { layout, heading, navLinks, linkHeading, link }: FooterData =
+    footer as FooterData;
```
Comment (haris-01) [inline components/Footer/index.tsx:15]: this doesn't seem right. why are we explicitly giving footer a type? even if are need to do that, it does not make sense to use it twice. 
 const { layout, heading, navLinks, linkHeading, link } =  footer as FooterData; should be fine

File: components/Footer/index.tsx [hunk 8]
```diff
@@ -1,5 +1,76 @@
...
+
+  const companySection = layout?.[0];
+  const contactSection = layout?.[1];
+  const imageSection = layout?.[2];
+  const textSection = layout?.[3];
+
+  const renderRichText = (richTextObject: any) =>
```
Comment (haris-01) [inline components/Footer/index.tsx:22]: please use proper type here instead of any

File: components/Footer/index.tsx [hunk 9]
```diff
@@ -1,5 +1,76 @@
...
+    richTextObject?.root?.children
+      ?.map((child: any) => child.children[0]?.text)
+      .join(" ") || "";
+
+  return (
+    <footer className="bg-zinc-200 pt-10 md:pt-14">
+      <div className="flex flex-col md:flex-row px-20  lg:flex-nowrap md:flex-wrap">
```
Comment (haris-01) [inline components/Footer/index.tsx:29]: we should create reusable component for flex container and flex children with basicnstyling which can also accept other classes as well. maybe shadCN already has something like that.

File: components/ui/button.tsx [hunk 10]
```diff
@@ -19,10 +19,12 @@ const buttonVariants = cva(
           "bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80",
         ghost: "hover:bg-accent hover:text-accent-foreground",
         link: "text-primary underline-offset-4 hover:underline",
+        custom: "bg-blue-900 text-white shadow-sm hover:bg-blue-800",
```
Comment (haris-01) [inline components/ui/button.tsx:22]: can we use a more descriptive name for this varient?

File: components/Footer/CompanyCard.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+const profileText = "Company Deck";
+const profileSize = "PDF, 4 MB";
+
+const CompanyCard: React.FC<CompanyCardProps> = ({
+  footerImage,
+  text,
+  buttonData,
+}) => {
+  if (!footerImage || !text || !buttonData) return null;
```
[the remark this answers is not in the document]
Comment (SumairaRasheed) [reply components/Footer/CompanyCard.tsx:23]: done

File: components/Footer/CompanyCard.tsx [hunk 12]
```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import Image from "next/image";
+import Link from "next/link";
+import { FaLongArrowAltRight } from "react-icons/fa";
+import { TbArrowBarToDown } from "react-icons/tb";
+import { Button } from "@/components/ui/button";
+import { ImageType, ButtonData } from "@/src/types/footerTypes";
```
[the remark this answers is not in the document]
Comment (SumairaRasheed) [reply components/Footer/CompanyCard.tsx:7]: done

File: components/Footer/ContactInfoCard.tsx [hunk 13]
```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { IoLocationSharp } from "react-icons/io5";
+import { MdEmail } from "react-icons/md";
+import { FaPhone } from "react-icons/fa";
+import { Layout } from "@/src/types/footerTypes";
+
+type ContactInfoProps = {
+  icon: React.ReactNode;
+  title: string;
+  content: string;
```
[the remark this answers is not in the document]
Comment (SumairaRasheed) [reply components/Footer/ContactInfoCard.tsx:10]: resolved

File: components/Footer/ContactInfoCard.tsx [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+        <div className="text-blue-900">{icon}</div>
+        <h5 className="font-bold text-sm text-blue-900">{title}</h5>
+      </div>
+    )}
+    <p className="lg:text-xs md:text-sm lg:ml-11 md:ml-11 pb-2">{content}</p>
+  </div>
+);
+
+const ContactInfoCard: React.FC<ContactInfoCardProps> = ({ layout }) => {
+  const { address, email, contactNumber, Heading } = layout;
```
[the remark this answers is not in the document]
Comment (SumairaRasheed) [reply components/Footer/ContactInfoCard.tsx:36]: changed

File: components/Footer/index.tsx (hunk 6 above)
Comment (SumairaRasheed) [reply components/Footer/index.tsx:11]: done

File: components/Footer/index.tsx (hunk 7 above)
Comment (SumairaRasheed) [reply components/Footer/index.tsx:15]: resolved

File: components/Footer/index.tsx (hunk 8 above)
Comment (SumairaRasheed) [reply components/Footer/index.tsx:22]: done

File: components/ui/button.tsx (hunk 10 above)
Comment (SumairaRasheed) [reply components/ui/button.tsx:22]: I am going to use "brand" name for this variant.

File: components/Footer/TechnologiesCard.tsx (hunk 3 above)
Comment (SumairaRasheed) [reply components/Footer/TechnologiesCard.tsx:24]: done

File: components/Footer/TechnologiesCard.tsx (hunk 5 above)
Comment (SumairaRasheed) [reply components/Footer/TechnologiesCard.tsx:28]: done

File: components/Footer/TechnologiesCard.tsx (hunk 4 above)
Comment (SumairaRasheed) [reply components/Footer/TechnologiesCard.tsx:21]: done

File: components/Footer/index.tsx (hunk 9 above)
Comment (SumairaRasheed) [reply components/Footer/index.tsx:29]: done

File: components/Footer/CompanyCard.tsx [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+const profileSize = "PDF, 4 MB";
+
+const CompanyCard: React.FC<CompanyCardProps> = ({
+  footerImage,
+  text,
+  buttonData,
+}) => {
+  return (
+    <div className="bg-stone-200  lg:max-w-80 pr-4">
+      <div>
```
Comment (Sheraz-Ilyas) [inline components/Footer/CompanyCard.tsx:25]: Are we safe to remove this extra `<div>`?

Comment (SumairaRasheed) [reply components/Footer/CompanyCard.tsx:25]: Yes , I have removed .

**Pull request merged**